### ISSUE TITLE
Hook keeps looping

### ISSUE THREAD

**Issue by kosmiq** [bdkjones/CodeKit2#596]
So I just started out using a Hook for a current project. The hook is supersimple, it just runs "grunt grunticon" (https://github.com/filamentgroup/grunticon)

![ck_hooks](https://cloud.githubusercontent.com/assets/322759/11555632/04a62742-99a2-11e5-9d8e-94da4609971a.png)

I have tried other settings to trigger the hook than those above, like the default, looking for minified files and so (but all JS-files).

What happens is that when I save "scripts.js" it gets processed as it should by CodeKit, minified and saved just fine. The hook will launch "grunt grunticon" (it does what it should) and generate all its stuff, including the file "grunticon.loader.js" which I also prepend to scripts.js.

I guess this is where things gets... Weird.

So in short:
- CodeKit runs and concatenate files and scripts.
- It executes "grunt grunticon" in the project folder
- Grunticon will generate grunticon.loader.js which is prepended into scripts.js
- It loops the hook, over and over again.

I guess it is because I save the script file, it runs the hook, a prepended file is changed, so CodeKit does it thing with uglify, concatetate and minify, and runs the hook again and it all starts over.

Is there a simple fix for this or should I simply stick to having it run on whenever style.css changes or whatever? I am aware that I might be seriously stupid here prepending a file that's created by a hook...

**Comment by bdkjones** [comment]
Ha, yea, you've set up an infinite loop. `script.js` is saved and CodeKit responds to that event by processing the file. At the end of processing, you have a Hook that modifies a file you prepend into `script.js`. CodeKit sees that file change, walks the dependency graph and says, "Script.js needs to be compiled because something it imports changed." 

You'll need to restructure this so that the loop isn't created; there's really no way for CodeKit to know that the change to the dependent file happened as a result of a Hook running on the first iteration of processing.

**Comment by kosmiq** [comment]
So I changed the hook to run when saving style.scss instead. Problem solved of course.
